tests/testthat/test-sce_to_anndata.R

```diff
@@ -20,7 +20,16 @@ metadata(sce) <- list(
   sample_metadata = data.frame(
     library_id = "library1",
     sample_id = "sample1"
-  )
+  ),
+  test_dataframe = tibble::tibble(
+    "char_col" = c("name1", NA),
+    "num_col" = 1:2,
+    "na_col" = c(NA, NA),
+    "list_col" = list(a = 1, b = "2")
+  ),
+  # these should get removed:
+  metadata_list = list(value = "value"),
+  metadata_DataFrame = DataFrame(head(iris))
```

Comment:
We _could_ convert `DataFrames` to `data.frames`, so long as there are not repeated rownames (which there should not be for any of our usages. In the one case that we currently have (as far as I am aware), `singler_results`, this would result in expanding the internal matrix to multiple columns, which I'm not necessarily upset by. I'm also not upset by removing it, so maybe up to you.

Reply:
>  To implement this, we should check if each DataFrame being expanded results in duplicate names, and don't include it if so.

Nope! Turns out `as.data.frame()` is running `make.unique()` under the hood, so this just works. I placed it as the first step in wrangling so if it got expanded to something problematic, that should get removed/fixed.